src/components/DynamicFormContainer/DynamicForm/CustomTemplates/CustomFileWidget.tsx

```diff
@@ -0,0 +1,190 @@
+import React, { FunctionComponent, useCallback, useState } from "react";
+import { useDropzone } from "react-dropzone";
+import { showInKB } from "../../../../common/utils";
+import { Button } from "../../../../UI/Button";
+import { SvgIcon, SvgIconPaperClip, SvgIconX } from "../../../../UI/SvgIcon";
+
+interface FileUploadType {
+  dataURL: string;
+  name: string;
+  size: number;
+  type: string;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const CustomFileWidget: FunctionComponent<any> = ({
+  onChange,
+  multiple,
+  options,
+  disabled,
+}) => {
+  const [uploadedFiles, setUploadedFiles] = useState<FileUploadType[]>([]);
+  const [fileSizeError, setFileSizeError] = useState(false);
+
+  const maxSize = 20000000;
+
+  const onDrop = useCallback(
+    (files: File[]) => {
+      let totalSize = uploadedFiles.reduce((acc, current) => acc + current.size, 0);
+      files.forEach((file: File) => {
+        totalSize += file.size;
+      });
+
+      if (totalSize <= maxSize) {
+        setFileSizeError(false);
+
+        processFiles(files).then((info: FileUploadType[]) => {
+          const allFiles = uploadedFiles.concat(info);
+          setUploadedFiles(allFiles);
+
+          const dataValue = allFiles.map((fileInfo: FileUploadType) => fileInfo.dataURL);
+          onChange(dataValue);
```

Comment:
Is there a replicated state somewhere else? 
If there is, you need to remove one. At any one point in time, a set of data should only be tracked at one component level, if you have replication, there is the problem of trying to sync it. 

However, since we are removing this from the json form, you will want to manage the entire attachment state at the context level, as oppose to having it's own internal state.

Reply:
ok got it